### PR TITLE
Modular computer improvements plus FUNCTIONING VIRUS

### LINKED ISSUE #885: AICs cannot interact with the home screen of modular computers
#### Description of issue
AICs when opening a console that is not in view of their **core** cannot access the UI if the computer is on the home screen, but, can access already open programs and even enable the computers.

Normally AICs can access consoles and laptops(when open) left around the site and interact with them.
The current behavior makes AICs unable to access consoles that are not in the visual range of the AI's core.

#### Steps to reproduce
\>Spawn as an AIC
\>Select a modular computer that is not in view of your core, like the ones in the control room. (You may also close your window blast \>doors to attempt to open the consoles in the observation room)
\>After the computer is activated, you cannot open its home page interface.

You may also test the consoles next to the AIC's chambers, those can be accessed normally when the blast doors are open and there's a line of sight with the AI's core.

#### Length of time in which bug has been known to occur
Roughly after the first TGUI update.

#### Issue bingo

- [x] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/foundation-19/foundation-19/issues)

### LINKED ISSUE #1031: Nanoword can't load files
#### Description of issue

You can save files, but you can't load files.

#### Difference between expected and actual behavior

Save a file, close it, reload it. Except you cant.

#### Steps to reproduce

Use a laptop, open up nanoword. Write something, save it, make a new file, then try to load the old file.

#### Length of time in which bug has been known to occur

No clue

#### Client version, Server revision & Game ID

Client Version: 514
Server Revision: Revision Unknown
Game ID: cnc-dhm1
Current map: Foundation Site 53

#### Issue bingo

- [x] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/foundation-19/foundation-19/issues)

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

General changes to modular computer system, hopefully should work a little better although to be honest it's still pretty bad.
Changes mainly were done to support new DIGITAL VIRUS system. Download files without protection (or open shady email attachments) to get a virus that general screws over your computer.

Fixes #885. Adds additional check to AI interaction; the computer must have an internet connection.
Fixes #1031.

## Why It's Good For The Game

Considering how much of SCP revolves around electronics, modular computers are probably going to be an important part of the future of SCP13. Adding content like digital viruses and such makes it easier to add more complex systems like digital SCPs, cybersecurity, and so on.

## Potential Fixes/Improvements

Modular computers are still a hot mess right now; inconsistent GUIs, buggy code, functionless files, and so on. Maybe not the highest priority, but there does need to be improvement sometime.

While the digital virus is a pretty cool feature, there are loads of ways to improve it (e.g. separate types of viruses, allowing antagonistic players to program their own viruses, more ways to spread viruses as well as ways to prevent that spread, more interesting effects that viruses can do, etc). I tried to leave behind a framework for future changes.

## Changelog

:cl:
add: New virus mechanics! Be careful when downloading without protection (or with shady emails), you might just end up with a bricked computer...
add: A computer needs to have internet access for AI to interact with it.
del: Docking management programs deleted.
balance: You no longer need access to run programs, only download them. Perhaps try stealing people's computers and running programs that you couldn't obtain otherwise?
fix: Loads of general modular computer fixes and improvements.
fix: Antag uplink program no longer meta-ble.
spellcheck: A couple modcomp-related lore improvements.
fix: AI not being able to interact with computer fixed.
fix: Nanoword not saving files correctly fixed.
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
I believe the SCP internet is referred to as SCiPNet, so you should probably change references to the internet with that.

**Comment by aidan-bowman** [comment]
> I believe the SCP internet is referred to as SCiPNet, so you should probably change references to the internet with that.

fixed

P.S. this can be integrated with my event PR, so if possible don't merge both at the same time.

**Comment by TheDarkElites** [comment]
Imagine forgetting to capitalize the n... :)

**Comment by TheDarkElites** [comment]
> Imagine forgetting to capitalize the n... :)

Actually nvm I think net is lowercase unless its .NET

File: code/modules/modular_computers/computers/modular_computer/core.dm [hunk 1]
```diff
@@ -220,7 +220,7 @@
 		return
 
 	if(P.requires_ntnet && !get_ntnet_status(P.requires_ntnet_feature)) // The program requires SCiPnet connection, but we are not connected to SCiPnet.
-		to_chat(user, SPAN_DANGER("\The [src]'s screen shows \"NETWORK ERROR - Unable to connect to SolNet. Please retry. If problem persists contact your system administrator.\" warning."))
+		to_chat(user, SPAN_DANGER("\The [src]'s screen shows \"NETWORK ERROR - Unable to connect to SCiPnet. Please retry. If problem persists contact your system administrator.\" warning."))
```
Comment (Tsurupeta) [inline code/modules/modular_computers/computers/modular_computer/core.dm:223]: Why scip

File: code/modules/modular_computers/file_system/program.dm [hunk 2]
```diff
@@ -120,7 +120,7 @@
 		if(3)
 			ntnet_speed = NTNETSPEED_ETHERNET
 
-// Check if the user can run program. Only humans can operate computer. Automatically called in run_program()
```
Comment (Tsurupeta) [inline code/modules/modular_computers/file_system/program.dm:123]: The proc is "can_run" but the comment is about downloading shit???

File: code/modules/modular_computers/computers/modular_computer/core.dm (hunk 1 above)
Comment (aidan-bowman) [reply code/modules/modular_computers/computers/modular_computer/core.dm:223]: It's just what the Foundation uses, it's pretty common on the wiki.

File: code/modules/modular_computers/file_system/program.dm (hunk 2 above)
Comment (aidan-bowman) [reply code/modules/modular_computers/file_system/program.dm:123]: fixed

File: code/modules/modular_computers/computers/modular_computer/core.dm (hunk 1 above)
Comment (TheDarkElites) [reply code/modules/modular_computers/computers/modular_computer/core.dm:223]: Also im pretty sure its already referenced as that somewhere in our codebase

**Review by Tsurupeta — APPROVED** [review]
I'm tired of being asked to review this piece of "good code", therefore I will merge it now and revert it in some (un)foreseeable future.

**Pull request merged**